packages/devkit/src/utils/convert-nx-executor.ts

```diff
@@ -36,29 +36,46 @@ export function convertNxExecutor(executor: Executor) {
           builderContext.target.target
         ];
     }
-    return from(toPromise(executor(options, context))).pipe(
-      map((output) => {
-        if (!output) {
-          return {
-            success: true,
-          };
-        } else {
-          return {
-            ...output,
-            success: true,
-          };
-        }
-      })
-    );
+    return toObservable(executor(options, context));
   };
   return require('@angular-devkit/architect').createBuilder(builderFunction);
 }
 
-async function toPromise(promiseOrAsyncIterator: any): Promise<any> {
-  if (typeof promiseOrAsyncIterator.then === 'function')
-    return promiseOrAsyncIterator;
-  let q;
-  for await (q of promiseOrAsyncIterator) {
+function toObservable<T extends { success: boolean }>(
+  promiseOrAsyncIterator: Promise<T> | AsyncIterableIterator<T>
+): Observable<T> {
+  if (typeof (promiseOrAsyncIterator as any).then === 'function') {
+    return from(promiseOrAsyncIterator as Promise<T>);
+  } else {
+    return new Observable((subscriber) => {
+      let asyncIterator = promiseOrAsyncIterator as AsyncIterableIterator<T>;
+      let prev: IteratorResult<T>;
+
+      function recurse(iterator: AsyncIterableIterator<T>) {
+        iterator
+          .next()
+          .then((result) => {
+            if (!result.done) {
+              subscriber.next(result.value);
+              prev = result;
```

Comment:
this `prev` variable looks like to be not used anywhere

Reply:
Thanks! :+1: